### PR TITLE
Change Reset switch to hard reset, trigger post_connect on boot and reset reconnects

### PR DESCRIPTION
**Pull request by drc38**
To help if HA/charger has a network error when starting

### CODE REVIEW

File: custom_components/ocpp/api.py [hunk 1]
```diff
@@ -566,8 +566,9 @@ async def stop_transaction(self):
             _LOGGER.debug("Failed with response: %s", resp.status)
             return False
 
-    async def reset(self, typ: str = ResetType.soft):
+    async def reset(self, typ: str = ResetType.hard):
         """Soft reset charger unless hard reset requested."""
```
Comment (lbbrhzn) [inline custom_components/ocpp/api.py:570]: Hard reset charger unless soft reset requested

File: custom_components/ocpp/api.py [hunk 2]
```diff
@@ -867,8 +868,9 @@ def on_boot_notification(self, **kwargs):
             om.charge_point_serial_number.name, None
         )
 
-        asyncio.create_task(self.async_update_device_info(kwargs))
+        self.hass.async_create_task(self.async_update_device_info(kwargs))
         self.hass.async_create_task(self.central.update(self.central.cpid))
+        self.hass.async_create_task(self.post_connect())
```
Comment (lbbrhzn) [inline custom_components/ocpp/api.py:873]: are these tasks executed sequentially or in parallel?

Comment (drc38) [reply custom_components/ocpp/api.py:873]: In parallel, the Hass create tasks wraps the asyncio function. Figure its best to use Hass one.

Comment (drc38) [reply custom_components/ocpp/api.py:872]: Realised this could lead to loop, see new commit

**Review by lbbrhzn — APPROVED** [review]

**Pull request merged**